ble_app.h

```diff
@@ -0,0 +1,427 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2019 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef BLE_APP_H_
+#define BLE_APP_H_
+
+#include "pretty_printer.h"
+#include "ble/BLE.h"
+#include "ChainableGapEventHandler.h"
+#include "events/mbed_events.h"
+#include "platform/Callback.h"
+#include "platform/NonCopyable.h"
+
+static const uint16_t MAX_ADVERTISING_PAYLOAD_SIZE = 50;
+
+/**
+ * Handle initialization and shutdown of the BLE Instance.
+ * It will also run the  event queue and call your post init callback when everything is up and running.
+ */
+class BLEApp : private mbed::NonCopyable<BLEApp>, public ble::Gap::EventHandler
+{
+public:
+    /**
+     * Construct a BLEApp from an event queue and a ble interface.
```

Comment:
No event queue is required to construct a `BLEApp`.
 
```suggestion
     * Construct a BLEApp from a BLE instance.
```